cosign/README.md

```diff
@@ -0,0 +1,248 @@
+    - Identity Log: An online provided record, that maps verified identities to their associated signing keys. Such logs help establish trust by providing a history of key associations.
+    - Artifact Log: An online/offline provided record that contains metadata about artifacts, such as when and by whom they were signed. This ledger aids in auditing and tracing the provenance of digital artifacts.
+    - Timestamps: Time is a critical component of Sigstore. It’s used to verify that a short-lived certificate issued by Fulcio was valid at a previous point, when the artifact was signed.
+
+## Scenarios
+
+`ratify-verifier-go` supports multiple verification scenarios based on different signing methods and trust sources. These scenarios can be categorized into the following main types.
+
+| Scenario             | Purpose                                                                | Use Case                                                                    |
+|-----------------------------------|------------------------------------------------------------------------|-----------------------------------------------------------------------------|
+| **Keyless Verification**          | Verifies signatures against Rekor, trust material (signed certificate timestamp) from the CT log, and certificates that chain up to Fulcio. | CI/CD pipelines or automated workflows where no private key management is needed. |
+| **Key-Based Verification**        | Verifies signature using a known **public key**.                        | Using an on-disk public key, or a key provided by other providers (e.g., "KMS provider") |
+
+### Cosign Library Keyless Verify
+
+To verify a keyless signed Cosign signature, you need to verify the following trust chain:
+
+- Check if the artifact matches what is described in the signature
+- Check if the OIDC identify (subject & issuer) matches what is described in the signature
+- Check if the certificate and the timestamp in signature are valid against Fulcio and Rekor
+- The root trust of Fulcio and Rekor is sourced from https://tuf-repo-cdn.sigstore.dev/
+- The root trust of https://tuf-repo-cdn.sigstore.dev/ is from the Cosign binary
+
+> [!NOTE]
+> With a short-lived certificate with timestamping, the trust has been completely shifted to Fulcio and Rekor of the Sigstore.
+> To resolve the TOFU issue of the TUF, Cosign pins the trust by embedding the current latest `root.json`, which contains the public key of the root role of the TUF, into the Cosign binary at the build time.
+
+#### Keyless Verify Input
+
+1. The Artifact to Verify
+    - The OCI subject.
+
+2. Verification Materials
+    - Identity Parameters: Restrict verification to specific OIDC identities. Valid values include email address, DNS names, IP addresses, and URIs.
+    - OIDC Issuer Parameters: Ensure the certificate was issued by a specific OIDC provider, without Fulcio roots (for BYO PKI).
```

Comment:
The phrase "without Fulcio roots (for BYO PKI)" requires rewording. Otherwise, it is confusing.

Reply:
update.